tidy3d/components/monitor.py

```diff
@@ -628,6 +628,21 @@ class AbstractFieldProjectionMonitor(SurfaceIntegrationMonitor, FreqMonitor):
         "in the far field of the device.",
     )
 
+    interval_space: Tuple[
+        pydantic.PositiveInt, pydantic.PositiveInt, pydantic.PositiveInt
+    ] = pydantic.Field(
+        (1, 1, 1),
+        title="Spatial interval",
+        description="Number of grid step intervals at which near fields are recorded for "
+        "projection to the far field, along each direction. If equal to 1, there will be no "
+        "downsampling. If greater than 1, the step will be applied, but the last point of the "
+        "monitor grid is always included. Using values greater than 1 can help speed up "
+        "server-side far field projections with minimal accuracy loss, especially in cases "
+        "where it is necessary for the grid resolution to be high for the FDTD simulation, "
+        "but such a high resolution is unnecessary for the purpose of projecting the recorded "
```

Comment:
so the interval space do not propagate to `projection_surfaces`? which I guess means that this will not be applied for user-side near2far? If I remember correctly, we had a way to downsample these? does it make sense to unify the downsampling somehow (like if interval space specified here, downsample when doing user-side projections too?)

Reply:
For client-side projections, the near fields are recorded via a `FieldMonitor` which already supports `interval_space`-based downsampling. Then on top of that, yeah, the user can further downsample based on points per wavelength in the client-side field projector. `interval_space` in the `FieldProjection*Monitor` when used client-side will have no effect, because in that case the projection monitor only defines the far field points, not the near field sources. I'm not sure if there's a good way to unify this, because for client-side projection, the `FieldProjection*Monitor` doesn't even need to be defined at simulation time, but can be defined afterwards just for the projection, in which case there's no way for its `interval_space` to propagate to the solver

Reply:
Ok makes sense thanks.